org.scala-ide.sdt.debug.expression.tests/src/org/scalaide/debug/internal/expression/remote/WithoutProjectSetTest.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2015 Contributor. All rights reserved.
+ */
+package org.scalaide.debug.internal.expression
+package remote
+
+import Names.Java
+import Names.Scala
+import org.junit.Ignore
+import org.junit.Test
+import org.junit.BeforeClass
+import org.junit.Assert._
+import TestValues.ValuesTestCase._
+import org.scalaide.debug.internal.expression.BaseIntegrationTest
+import org.scalaide.debug.internal.expression.BaseIntegrationTestCompanion
+import org.scalaide.debug.internal.expression.EclipseProjectContext
+import org.scalaide.debug.internal.expression.ReflectiveCompilationFailedWithClassNotFound
+
+class WithoutProjectSetTest extends BaseIntegrationTest(WithoutProjectSetTest) {
+
+  @Test(expected = classOf[MethodTypeInferred])
+  def testMethodAsLambda(): Unit =
+    eval(""" list.map(alaMethod) """, list.map(alaMethod), Scala.::)
+
+  @Test
+  def objectAccess(): Unit =
+    expectReflectiveCompilationError(""" LibObject.id """)
+
+  @Test
+  def javaStaticMemberTest(): Unit =
+    expectReflectiveCompilationError("JavaLibClass.staticString")
+
+  @Test
+  def localValAccess(): Unit =
+    eval(""" list """, list, Scala.::)
+
+  @Test
+  def classValAccess(): Unit =
+    eval(""" outer """, outer, Java.String)
+
+  @Test
+  def localMethod(): Unit =
+    eval(""" alaMethod(1) """, alaMethod(1), Java.String)
+
+  @Test
+  def nestedMethod(): Unit =
+    expectReflectiveCompilationError(""" testFunction() """)
+
+  @Test
+  def typedLambda(): Unit =
+    eval(""" list.map( (i: Int)=> i + 1) """, list.map((i: Int) => i + 1), Scala.::)
+
+  @Test(expected = classOf[FunctionProxyArgumentTypeNotInferredException])
+  def lambda(): Unit =
+    eval(""" list.map( _.toString) """, list.map(_.toString), Scala.::)
+
+  @Test
+  def lambdaThatRequireType(): Unit =
+    expectReflectiveCompilationError(""" list.map( _ + 1) """)
+
+  @Test
+  def lambdaThatRequireReturnValue(): Unit =
+    expectReflectiveCompilationError(""" list.filter( _ > 0) """)
+
+  @Test
+  def typedLambdaThatRequireReturnValue(): Unit =
+    eval(""" list.filter(( _: Int) > 0) """, list.filter(_ > 0), Scala.::)
+
+  @Test
+  def expectReflectiveCompilationError(): Unit =
+    expectReflectiveCompilationError(""" list.map( _ + int) """)
+
+  @Test
+  def typedLambdaWithClosure(): Unit =
+    eval(""" list.map( (i: Int)=> i + int) """, list.map((i: Int) => i + 1), Scala.::)
+
+  @Ignore("TODO O-8791 - No class def found is thrown")
+  @Test
```

Comment:
Could you change `@Ignore` to `@Test(expected = classOf[NoClassDefFound]`)? That way if error changes in future (or is no longer there) we will get feedback.